index.js

```diff
@@ -347,6 +348,10 @@ class XiaomiRoborockVacuum {
       this.log.info('STA getDevice | FanSpeed: ' + this.device.property("fanSpeed"));
       this.log.info('STA getDevice | BatteryLevel: ' + this.device.property("batteryLevel"));
 
+      if (this.config.autoroom) {
+        this.getRoomMap();
```

Comment:
Mind the method is a promise, you need to `await` it or you'll get an 'unhandled promise exception'.
```suggestion
        await this.getRoomMap();
```

Reply:
yes you are totally right ... sorry i'm on medicin at the moment ... sleep the hole day ...